### PR TITLE
RTDB profiler could expose queueing delay

### PR DESCRIPTION
**Pull request by IanWyszynski**
This change is `npm test` clean.

### Description

This is the accompanying firebase command-line tool change that exposes this field, if it exists, and generates tables, json, or raw output for it (as we currently do for readSpeed, writeSpeed, and broadcastSpeed). The following operations should have a `pendingTime` associated with them:

- concurrent-connect
- concurrent-disconnect
- realtime-read
- realtime-write
- realtime-transaction
- realtime-update
- rest-read
- rest-write
- rest-update
- listener-unlisten

We elide profiler messages that do not contain the `pendingTime` field even when we expect that they should exist. This enables us to independently release the command-line tools and RTDB.
	 
### Scenarios Tested

I have tested running:

`firebase database:profile --project wyszynski-testing`

against the staging RTDB (which has been updated with the change to firebase-server that introduces `pendingTime`) and the production RTDB (which does not emit `pendingTime`). In the former case, I saw the `pendingTime` field in the output. In the latter case I did not. 

### Sample Commands

This change does not introduce any change to the set of flags the firebase CLI tool accepts. It only augments the output of `firebase database:profile`.

### CODE REVIEW

**Comment by IanWyszynski** [comment]
Corresponding firebase-server PR: https://github.com/firebase/firebase-server/pull/2833
Proposed change to public docs: https://critique.corp.google.com/#review/245867448

File: src/profileReport.js [hunk 1]
```diff
@@ -173,8 +221,10 @@ ProfileReport.prototype.processOperation = function(data) {
   this.state.opCount++;
   switch (data.name) {
     case "concurrent-connect":
+      this.collectConnect(data, path);
       break;
     case "concurrent-disconnect":
+      this.collectDisconnect(data, path);
       break;
```
[the remark this answers is not in the document]
Comment (IanWyszynski) [reply src/profileReport.js]: This corresponds to the `concurrent-disconnect` profiler record type.

**Comment by IanWyszynski** [comment]
Per an offline conversation with @rockwotj, I'll be rolling the `pendingTime` metric into the firebase-tools speed report.

**Comment by bkendall** [comment]
Looks fine to me

File: src/profileReport.js [hunk 2]
```diff
@@ -349,7 +383,7 @@ ProfileReport.prototype.renderIncomingBandwidth = function() {
 };
 
 ProfileReport.prototype.renderOperationSpeed = function(pureData, hasSecurity) {
-  var head = ["Path", "Count", "Average"];
+  var head = ["Path", "Count", "Average Execution Speed", "Pending Count", "Average Pending Time"];
```
Comment (fredzqm) [inline src/profileReport.js]: "Pending Count" may get confusing to our customers.

Comment (IanWyszynski) [reply src/profileReport.js]: I could omit that column altogther.

**Pull request merged**